### PR TITLE
ECDC-2518: Restore setpanel information to default when experiment finished

### PR DESCRIPTION
**Pull request by ebadkamil**

### CODE REVIEW

**Review by kmurica — APPROVED** [review]
Works fine. LGTM.

**Comment by kmurica** [comment]
Some conflict needs to be resolved before merge though.

File: nicos_ess/gui/panels/setup_panel.py [hunk 1]
```diff
@@ -388,6 +393,7 @@ def on_queryDBButton_clicked(self):
 
             if result:
                 if len(result) != 1:
+                    # chooseProposal is not defined
```
Comment (mattclarke) [inline nicos_ess/gui/panels/setup_panel.py:396]: This deserves a bug ticket rather than a comment

**Comment by mattclarke** [comment]
The conflicts are probably due to the gui stuff moving. If the merge conflicts look complicated let me know and I'll sort them out.

**Comment by ebadkamil** [comment]
The conflicts stayed even after merging main, due to github outage sometime ago. Now looks ok.

File: nicos_ess/gui/panels/setup_panel.py (hunk 1 above)
Comment (ebadkamil) [reply nicos_ess/gui/panels/setup_panel.py]: Done. ECDC-2559.

File: nicos_ess/gui/panels/setup_panel.py [hunk 2]
```diff
@@ -191,6 +191,11 @@ def _update_proposal_info(self):
         samples_dict = {} if self.hide_samples \
             else self.client.eval('Exp.sample.samples', {})
 
+        # FinishExperiment() sets the samples to [0: {'name':''}]
+        # EssSample device may fix this.
+        if len(samples_dict) == 1 and 'sample_name' not in samples_dict[0]:
```
Comment (mattclarke) [inline nicos_ess/gui/panels/setup_panel.py:196]: This is a hack!

Perhaps it is possible to clear the samples in `on_experiment_finished` instead?
Or, even better, lower down in the EssExperiment class?

Comment (ebadkamil) [reply nicos_ess/gui/panels/setup_panel.py:196]: Yes. I agree it's a hack.  I can also do something like 
`self.client.run("Exp.sample.clear()")` here in `on_experiment_finished` . But I would go for `EssExperiment` class. for this PR, we can either keep it like it is or use `self.client.run("Exp.sample.clear()")`.
What do you think?

Comment (mattclarke) [reply nicos_ess/gui/panels/setup_panel.py:196]: How about you don't fix it here (i.e. remove the hack), merge this PR, then fix it properly in a new PR?

Comment (ebadkamil) [reply nicos_ess/gui/panels/setup_panel.py]: Alright. I will create a new ticket

Comment (ebadkamil) [reply nicos_ess/gui/panels/setup_panel.py]: New ticket ECDC-2560.

File: nicos/clients/flowui/mainwindow.py [hunk 3]
```diff
@@ -60,6 +60,9 @@ def __init__(self, log, gui_conf, viewonly=False, tunnel=''):
         self.getPanel('Experiment setup').exp_proposal_activated.connect(
             self.getPanel('Finish experiment').on_new_experiment_proposal
         )
+        self.getPanel('Finish experiment').experiment_finished.connect(
+            self.getPanel('Experiment setup').on_experiment_finished
+        )
```
Comment (mattclarke) [inline nicos/clients/flowui/mainwindow.py:65]: Just stumbled across this while looking at a bug in the other version: `self.client.experiment.connect(self.on_client_experiment)`
This event triggers when an experiment is started and **finished**, so we don't need this new signal, we can just add `self.client.experiment.connect(self.on_client_experiment)` to `initialise_connection_status_listeners` 🎉

Comment (mattclarke) [reply nicos/clients/flowui/mainwindow.py:65]: Perhaps

Comment (ebadkamil) [reply nicos/clients/flowui/mainwindow.py]: Please see if the new changes are correct.

**Pull request merged**